Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -30,31 +47,293 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
+        if lastActiveMembers.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder4,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        var urls = lastActiveMembers.map(\.imageURL)
+        
+        if urls.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder3,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        // We show a combination of at max 4 images combined
+        urls = Array(urls.prefix(4))
+        
+        guard !SystemEnvironment.isTests else {
+            // When running tests, we load the images synchronously
+            let images = urls.map { UIImage(data: try! Data(contentsOf: $0!))! }
+           
+            let combinedImage = createMergedAvatar(from: images) ?? appearance.images.userAvatarPlaceholder2
+            
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: combinedImage,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        loadAvatarsFrom(urls: urls, channelId: channel.cid) { [weak self] avatars, channelId in
+            guard let self = self, channelId == self.content.channel?.cid else { return }
+            
+            let combinedImage = self.createMergedAvatar(from: avatars) ?? self.appearance.images.userAvatarPlaceholder2
+            
+            self.presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: combinedImage,
+                preferredSize: .avatarThumbnailSize,
+                components: self.components
+            )
+        }
+    }
+    
+    /// Loads avatars for the given URLs
+    /// - Parameters:
+    ///   - urls: The avatar urls
+    ///   - channelId: The channelId of the channel
+    ///   - completion: Completion that gets called with an array of `UIImage`s when all the avatars are loaded
+    open func loadAvatarsFrom(
+        urls: [URL?],
+        channelId: ChannelId,
+        completion: @escaping ([UIImage], ChannelId)
+            -> Void
+    ) {
+        var placeholderAvatars: [UIImage] = []
+        
+        var placeholderImages = [
+            appearance.images.userAvatarPlaceholder1,
+            appearance.images.userAvatarPlaceholder2,
+            appearance.images.userAvatarPlaceholder3,
+            appearance.images.userAvatarPlaceholder4
+        ]
+        
+        var avatarUrls: [URL] = []
+        
+        for url in urls.prefix(4) {
```

Comment:
.prefix(4) - magic number

can be moved to private enum
```Swift
private enum Configuration {
   static let httpPrefixSymbolsCount = 4
}
```

and use as `Configuration.httpPrefixSymbolsCount`

Reply:
1. I hope the number `4` here represents the maximum number of images that we merge to create a new avatar. I don't understand why it should be named as `httpPrefixSymbolsCount`.

1. Wouldn't a private let be enough? 

```swift
private let maxNoOfImagesToBeMergedToGenerateAvatar = 4
```
Thanks!

Reply:
1. thats why its called 'magic number', cause for reader is very hard to guess why this constant is placed this place :)
2. yes, private let or static let inside private enum is fine, we could keep it without enum as long its only one variable

Reply:
Sure, I understand the magic number part. I was to trying to understand why you named it as `httpPrefixSymbolsCount`.

Reply:
✅